### ISSUE TITLE
Dropdown behaviour inconsistent & incorrect tags displayed

### ISSUE THREAD

**Issue by techie42** [2amigos/yii2-taggable-behavior#25]
When using a form, the input field for tags is populated correctly.  However, clicking in the input field does not display the available tags.  Only once I begin to type do any matching tags appear in a dropdown.

If I then delete the typed characters without creating a new tag, then move focus away from the input field, then click back in the input field, the dropdown immediately appears with the same matched tags displayed, even though I haven't typed anything yet.

This means that other tags that don't match are not displayed, misleading other users as they assume the displayed tags are now all of them instead of just those that matched text typed earlier (which was deleted from the input field).

My "SelectizeTextInput" widget is configured as follows:

delimiter => ',', 'create' => true, 'persist' => true, 'createOnBlur' => false, 'openOnFocus' => true, 'preload' => true

A dirty solution I've found is to edit InputWidget.php in the Selectize widget (I'm using the Taggable behaviour) as follows:

Around line 47:
change
`$this->clientOptions['load'] = new JsExpression("function (query, callback) { if (!query.length) return callback(); $.getJSON('$url', { query: encodeURIComponent(query) }, function (data) { callback(data); }).fail(function () { callback(); }); }");`

to
`$this->clientOptions['load'] = new JsExpression("function (query, callback) { $.getJSON('$url', { query: encodeURIComponent(query) }, function (data) { callback(data); }).fail(function () { callback(); }); }");`

By simply removing the check for the presence of the query, the dropdown appears *every* time I click in the input field, displaying *all* available tags.  As soon as I start typing, the displayed list narrows to match the text.  I believe the check for the query length interferes with the "preload" configuration option of Selectize, which is what I'm trying to use to ensure the intial single-click in the input field opens a dropdown with all tags available.

This is the behaviour I was expecting from the Taggable / Selectize setup, but maybe I've just got it all wrong :)

I've also found an issue in "Taggable.php" around line 113:
`return $this->asArray ? $items : implode(', ', $items);`

This should be changed to this:
`return $this->asArray ? $items : implode(',', $items);`

Basically, removing the extra space after the comma between the single-quotes - this interferes with the pattern matching, causing some tags to be displayed when they shouldn't, and others not to display because they don't match.

By the way, you guys build awesome controls ... *please* keep up the great work :)

**Comment by tonydspaniard** [comment]
Hi @techie42 thanks for your feedback... do you have time for quick pr?
